pkg/identityserver/end_device_registry.go

```diff
@@ -205,6 +205,21 @@ func (is *IdentityServer) updateEndDevice(ctx context.Context, req *ttnpb.Update
 		req.FieldMask = cleanFieldMaskPaths(ttnpb.EndDeviceFieldPathsNested, req.FieldMask, nil, []string{"activated_at"})
 	}
 
+	if ttnpb.HasAnyField(req.FieldMask.GetPaths(), "last_seen_at") {
+		err = is.store.Transact(ctx, func(ctx context.Context, st store.Store) (err error) {
+			dev, err = st.GetEndDevice(ctx, req.EndDevice.Ids, store.FieldMask{"last_seen_at"})
```

Comment:
Just write the value, no need to add these checks and round trips. The value can be eventually consistent.